src/Command/RecipesCommand.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex\Command;
+
+use Composer\Command\BaseCommand;
+use Composer\Factory;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Flex\InformationOperation;
+use Symfony\Flex\Lock;
+use Symfony\Flex\Recipe;
+
+class RecipesCommand extends BaseCommand
+{
+    /** @var \Symfony\Flex\Flex */
+    private $flex;
+
+    /** @var Lock */
+    private $symfonyLock;
+
+    public function __construct(/* cannot be type-hinted */ $flex)
+    {
+        $this->flex = $flex;
+        $this->symfonyLock = new Lock(getenv('SYMFONY_LOCKFILE') ?: str_replace('composer.json', 'symfony.lock', Factory::getComposerFile()));
```

Comment:
I think this can be passed in via the constructor. It's created inside `Flex::activate()` before that method initializes the commands. I'm not sure why it wasn't also passed directly to the `SyncRecipesCommand` (which also instantiates the `new Lock()` like this.